src/core/server/opensearch/opensearch_config.test.ts

```diff
@@ -82,7 +82,7 @@ test('set correct defaults', () => {
         "enabled": false,
         "maxPercentage": 1,
       },
-      "optimizedHealthcheckId": undefined,
+      "optimizedHealthcheck": undefined,
```

Comment:
What happen to existing 1.2 customer who forget to  update their dashboards.yml file from`opensearch.optimizedHealthcheckId` to `opensearch.optimizedHealthcheck.Id`

Reply:
OSD will fail to start because `opensearch.optimizedHealthcheckId` cannot be recognized. We can call this out in the release note that customers may need to update their yml. It may be a bit over engineering if we want to keep both `optimizedHealthcheckId` and `optimizedHealthcheck` and have `if/else` to use `optimizedHealthcheckId` if it is provided `optimizedHealthcheckId`.

Reply:
I've added this to the deprecated service so it will work and properly map the value from optimizedHealthcheckId -> optimizedHealthcheck.id and then log a warning message to use the new configuration

Reply:
@kavilla I missed the deprecated service part initially. Looks good!